pkg/util/cluster.go

```diff
@@ -19,8 +19,53 @@ import (
 const (
 	// NamespaceClusterLease is the namespace which cluster lease are stored.
 	NamespaceClusterLease = "karmada-cluster"
+	// KubeCredentials is the secret that contains mandatory credentials whether reported when registering cluster
+	KubeCredentials = "KubeCredentials"
+	// KubeImpersonator is the secret that contains the token of impersonator whether reported when registering cluster
+	KubeImpersonator = "KubeImpersonator"
+	// None is means don't report any secrets.
+	None = "None"
 )
 
+// RegistryClusterOption represents the option for RegistryCluster.
+type RegistryClusterOption struct {
```

Comment:
I like this struct.
Do you think `ClusterRegisterOption` would be better?

Reply:
> Do you think ClusterRegisterOption would be better?

OK